components/OrderPage/OrderPage.tsx

```diff
@@ -5,42 +5,65 @@ import style from './OrderPage.module.scss'
 import { MenuItemGroup } from '../../models/interfaces'
 import { FaWindowClose } from 'react-icons/fa'
 import LoaderComponent from '../verse/LoaderComponent'
-import useStoredCheckout from '../../hooks/useStoredCheckout'
-import { CHECKOUT_STORAGE_KEY, CURRENT_ORDERS_KEY } from '../../constants/localStorageConstants'
+import { MenegeableStorage } from '../../hooks/useStoredCheckout'
+import { CURRENT_ORDERS_KEY } from '../../constants/localStorageConstants'
 import useSimpleLocalStorage from '../../hooks/useStoredSentOrder'
 import sendOrderData from './sendOrderData'
 import { useRouter } from 'next/router'
+import { localStorageContext } from '../contexts/LocalStorageProvider'
+import { createManageableStorage } from '../../utils/localStorageHelper'
+import { generateID } from '../../utils/modelHelper'
 
 export default function OrderPage() {
    const router = useRouter()
 
-   // This one manages orders sent to shopkeeper
-   const { addItem } = useSimpleLocalStorage(CURRENT_ORDERS_KEY)
-
-   const { menuItems, toppings, sauces, menuItemOptions } = useContext(menuContext)
-
-   // This one manages orders yet to be sent to shopkeeper
-   const { storedData, manageableStorage, setLocalStorage, clearLocalStorage } =
-      useStoredCheckout<MenuItemGroup>(CHECKOUT_STORAGE_KEY)
-   // counting amount of each repeated order
-   const [counter, setCounter] = useState<number[]>(
-      manageableStorage.map((list) => list.ids.length)
+   const { menuItemGroups, addMenuItemGroup, removeMenuItemGroup, clearMenuItemGroups } =
+      useContext(localStorageContext)
+   const [manageableLocalStorage, setManageableLocalStorage] = useState(
+      createManageableStorage(menuItemGroups)
    )
+
    const priceWatcher = () => {
       let bill = 0
-      manageableStorage.forEach((unique, idx) => {
-         bill += unique.order.subTotal * counter[idx]
+      manageableLocalStorage.forEach((unique, idx) => {
+         bill += unique.order.subTotal * unique.ids.length
       })
```

Comment:
This can be simplified by using `reducer` function. Good opportunity to improve code skills. Search and apply reducer here. Ensure value is the same before and after the change from `forEach` to `reduce`.